### DISCUSSION TITLE
The issue regarding publishing on Kafka and RabbitMQ

### DISCUSSION THREAD

**Discussion by Tynab** [abpframework/abp#16036] [Q&A]
I use PublishAsync of KafkaDistributedEventBus and RabbitMqDistributedEventBus in 2 different services sharing the same source code, but it only publishes both ETOs to the same location (either Kafka or RabbitMQ) depending on the placement of AbpEventBus...Module in DependsOn. If AbpEventBusKafkaModule is placed after both ETOs, they will be published to Kafka, while if AbpEventBusRabbitMqModule is placed after them, both ETOs will be published to RabbitMQ. My issue is that I want ETOs to be published to Kafka for the service using KafkaDistributedEventBus, and to RabbitMQ for the service using RabbitMqDistributedEventBus. Can anyone help me solve this problem? Thank you.

**Comment by gterdem** [comment]
You can select to publish whichever eventbus you want by injecting the concrete type.

Ex: Instead of injecting `IDistributedEventBus`, you can directly inject the `RabbitMqDistributedEventBus` to publish.
